<launch>
  <arg name="density"  default="1"/>
  <arg name="scale"  default="1"/>
  <arg name="objects"  default="teapot cube"/>
  <node name="sphere1_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="$(arg density)"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="sphere1_points"/>
    <param name="link_name"  type="str"  value="sphere1_link"/>
    <remap from="cube_world_pub" to="sphere1_world_pub" />
    <remap from="cube_world" to="sphere1_world" />
    <remap from="cube_points" to="sphere1_points" />
    <remap from="cloud_pcd" to="sphere1_cloud_pcd" />
    <remap from="SpherePose_ros" to="Sphere1Pose_ros" />
  </node>
  <!-- <node name="sphere2_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="$(arg density)"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="sphere2_points"/>
    <param name="link_name"  type="str"  value="sphere2_link"/>
    <remap from="cube_world_pub" to="sphere2_world_pub" />
    <remap from="cube_world" to="sphere2_world" />
    <remap from="cube_points" to="sphere2_points" />
    <remap from="cloud_pcd" to="sphere2_cloud_pcd" />
    <remap from="SpherePose_ros" to="Sphere2Pose_ros" />
  </node> -->
  <!-- <node name="sphere3_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="$(arg density)"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="sphere3_points"/>
    <param name="link_name"  type="str"  value="sphere3_link"/>
    <remap from="cube_world_pub" to="sphere3_world_pub" />
    <remap from="cube_world" to="sphere3_world" />
    <remap from="cube_points" to="sphere3_points" />
    <remap from="cloud_pcd" to="sphere3_cloud_pcd" />
    <remap from="SpherePose_ros" to="Sphere3Pose_ros" />
  </node> -->
  <!-- <node name="sphere4_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="$(arg density)"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="sphere4_points"/>
    <param name="link_name"  type="str"  value="sphere4_link"/>
    <remap from="cube_world_pub" to="sphere4_world_pub" />
    <remap from="cube_world" to="sphere4_world" />
    <remap from="cube_points" to="sphere4_points" />
    <remap from="cloud_pcd" to="sphere1_cloud_pcd" />
    <remap from="SpherePose_ros" to="Sphere4Pose_ros" />
  </node>
  <node name="sphere5_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="$(arg density)"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="sphere5_points"/>
    <param name="link_name"  type="str"  value="sphere5_link"/>
    <remap from="cube_world_pub" to="sphere5_world_pub" />
    <remap from="cube_world" to="sphere5_world" />
    <remap from="cube_points" to="sphere5_points" />
    <remap from="cloud_pcd" to="sphere1_cloud_pcd" />
    <remap from="SpherePose_ros" to="Sphere5Pose_ros" />
  </node>
  <node name="sphere6_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="$(arg density)"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="sphere6_points"/>
    <param name="link_name"  type="str"  value="sphere6_link"/>
    <remap from="cube_world_pub" to="sphere6_world_pub" />
    <remap from="cube_world" to="sphere6_world" />
    <remap from="cube_points" to="sphere6_points" />
    <remap from="cloud_pcd" to="sphere1_cloud_pcd" />
    <remap from="SpherePose_ros" to="Sphere6Pose_ros" />
  </node>
  <node name="sphere7_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="$(arg density)"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="sphere7_points"/>
    <param name="link_name"  type="str"  value="sphere7_link"/>
    <remap from="cube_world_pub" to="sphere7_world_pub" />
    <remap from="cube_world" to="sphere7_world" />
    <remap from="cube_points" to="sphere7_points" />
    <remap from="cloud_pcd" to="sphere2_cloud_pcd" />
    <remap from="SpherePose_ros" to="Sphere7Pose_ros" />
  </node>
  <node name="teapot_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="100"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="teapot_points"/>
    <param name="link_name"  type="str"  value="teapot_link"/>
    <remap from="cube_world_pub" to="teapot_world_pub" />
    <remap from="cube_world" to="teapot_world" />
    <remap from="cube_points" to="teapot_points" />
    <remap from="cloud_pcd" to="teapot_cloud_pcd" />
    <remap from="SpherePose_ros" to="Teapot_ros" />
  </node> -->
  <node name="table_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="1"/>
    <param name="scale"  type="double"  value="5"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="table_points"/>
    <param name="link_name"  type="str"  value="table_link"/>
    <remap from="cube_world_pub" to="table_world_pub" />
    <remap from="cube_world" to="table_world" />
    <remap from="cube_points" to="table_points" />
    <remap from="cloud_pcd" to="table_cloud_pcd" />
    <remap from="SpherePose_ros" to="TablePose_ros" />
  </node>
  <!-- <node name="bunny_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="100"/>
    <param name="scale"  type="double"  value="10"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="bunny_points"/>
    <param name="link_name"  type="str"  value="bunny_link"/>
    <remap from="cube_world_pub" to="bunny_world_pub" />
    <remap from="cube_world" to="bunny_world" />
    <remap from="cube_points" to="bunny_points" />
    <remap from="cloud_pcd" to="bunny_cloud_pcd" />
    <remap from="SpherePose_ros" to="BunnyPose_ros" />
  </node> -->
  <node name="bottle_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="1"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="bottle_points"/>
    <param name="link_name"  type="str"  value="bottle_link"/>
    <remap from="cube_world_pub" to="bottle_world_pub" />
    <remap from="cube_world" to="bottle_world" />
    <remap from="cube_points" to="bottle_points" />
    <remap from="cloud_pcd" to="bottle_cloud_pcd" />
    <remap from="SpherePose_ros" to="BottlePose_ros" />
  </node>
  <node name="meat_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="1"/>
    <param name="scale"  type="double"  value="10"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="meat_points"/>
    <param name="link_name"  type="str"  value="meat_link"/>
    <remap from="cube_world_pub" to="meat_world_pub" />
    <remap from="cube_world" to="meat_world" />
    <remap from="cube_points" to="meat_points" />
    <remap from="cloud_pcd" to="meat_cloud_pcd" />
    <remap from="SpherePose_ros" to="MeatPose_ros" />
  </node>
  <node name="potato_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="1"/>
    <param name="scale"  type="double"  value="1000"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="potato_points"/>
    <param name="link_name"  type="str"  value="potato_link"/>
    <remap from="cube_world_pub" to="potato_world_pub" />
    <remap from="cube_world" to="potato_world" />
    <remap from="cube_points" to="potato_points" />
    <remap from="cloud_pcd" to="potato_cloud_pcd" />
    <remap from="SpherePose_ros" to="PotatoPose_ros" />
  </node>
  <node name="tomato_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="1"/>
    <param name="scale"  type="double"  value="100"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="tomato_points"/>
    <param name="link_name"  type="str"  value="tomato_link"/>
    <remap from="cube_world_pub" to="tomato_world_pub" />
    <remap from="cube_world" to="tomato_world" />
    <remap from="cube_points" to="tomato_points" />
    <remap from="cloud_pcd" to="tomato_cloud_pcd" />
    <remap from="SpherePose_ros" to="TomatoPose_ros" />
  </node>
  <node name="tomato2_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="1"/>
    <param name="scale"  type="double"  value="100"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="tomato2_points"/>
    <param name="link_name"  type="str"  value="tomato2_link"/>
    <remap from="cube_world_pub" to="tomato2_world_pub" />
    <remap from="cube_world" to="tomato2_world" />
    <remap from="cube_points" to="tomato2_points" />
    <remap from="cloud_pcd" to="tomato2_cloud_pcd" />
    <remap from="SpherePose_ros" to="Tomato2Pose_ros" />
  </node>
  <node name="veggie_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" >
    <param name="density"  type="double"  value="1"/>
    <param name="scale"  type="double"  value="100"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="veggie_points"/>
    <param name="link_name"  type="str"  value="veggie_link"/>
    <remap from="cube_world_pub" to="veggie_world_pub" />
    <remap from="cube_world" to="veggie_world" />
    <remap from="cube_points" to="veggie_points" />
    <remap from="cloud_pcd" to="veggie_cloud_pcd" />
    <remap from="SpherePose_ros" to="VeggiePose_ros" />
  </node>
  <node name="human_pub" pkg="human_detection" type="fake_Node.py" output = "screen"/>
  <!-- <node name="perceptual_map" pkg="convert_pcd" type="perceptual_map_v2.py" output="screen" /> -->
  <!-- <include file="$(find octomap_server)/launch/octomap_mapping.launch" />  -->
  <!-- <node name="sphere2_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" output="screen">
    <param name="density"  type="double"  value="$(arg density)"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="~frame_id"  type="str"  value="sphere2_points"/>
    <param name="link_name"  type="str"  value="sphere2_link"/>
    <remap from="cube_world_pub" to="sphere2_world_pub" />
    <remap from="cube_world" to="sphere2_world" />
    <remap from="cube_points" to="sphere2_points" />
    <remap from="cloud_pcd" to="sphere2_cloud_pcd" />
    <remap from="SpherePose_ros" to="SpherePose2_ros" />
  </node>
  <node name="cylinder_to_world" pkg="convert_pcd" type="obj_to_world_perception.py" output="screen">
    <param name="density"  type="double"  value="$(arg density)"/>
    <param name="scale"  type="double"  value="$(arg scale)"/>
    <param name="objects"  type="str"  value="$(arg objects)"/>
    <param name="frame_id"  type="str"  value="cylinder_points"/>
    <param name="link_name"  type="str"  value="cylinder_link"/>
    <remap from="cube_world_pub" to="cylinder_world_pub" />
    <remap from="cube_world" to="cylinder_world" />
    <remap from="cube_points" to="cylinder_points" />
    <remap from="cloud_pcd" to="cylinder_cloud_pcd" />
    <remap from="SpherePose_ros" to="CylinderPose_ros" />
  </node> -->
</launch>